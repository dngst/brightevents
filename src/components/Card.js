import styled from "styled-components";

const Card = styled.div`
  height: 100%;
  margin: 0 0 4%;
  border: 2px dashed ${(props) => props.theme.blue};
  text-align: left;
  padding: 0 4%;
  p {
    font-size: 1rem;
    line-height: 1.2;
  }
  .event-title {
    font-size: 1.2rem;
    text-transform: uppercase;
    font-family: sans-serif;
    a {
      color: ${(props) => props.theme.grey};
      text-decoration: underline;
    }
    span {
      color: ${(props) => props.theme.highlight};
      font-weight: bold;
    }
  }
  .event-cat {
    font-size: 0.9rem;
  }
  @media (max-width: 480px) {
    padding: 0;
    border: none;
    .event-title {
      span {
        display: none;
      }
    }
  }
`;

export default Card;
