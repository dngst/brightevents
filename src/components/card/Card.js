import styled from "styled-components";

const Card = styled.div`
  height: 100%;
  margin: 0 0 4%;
  border: 3px dashed #C1DBE3;
  text-align: left;
  padding: 0 4%;
  p {
    font-size: 1rem;
    line-height: 1.2;
  }
  .card-title {
    font-size: 1.2rem;
    text-transform: uppercase;
    a {
      color: #81523F;
      text-decoration: underline;
    }
    span {
      color: #CA1551;
    }
    @media (max-width: 480px) {
      span {
        display: none;
      }
    }
  }
`

export default Card;
