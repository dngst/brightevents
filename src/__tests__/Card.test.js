import { render } from "@testing-library/react";
import Card from "components/Card";

it("works", () => {
  const { container } = render(<Card />);
  expect(container.firstChild).toMatchSnapshot();
});

it("applies default styles", () => {
  const { container } = render(<Card />);
  expect(container.firstChild).toHaveStyleRule("text-align", "left");
  // expect(container.firstChild).toHaveStyleRule("font-size", "1rem");
  expect(container.firstChild).toHaveStyleRule("height", "100%");
});
