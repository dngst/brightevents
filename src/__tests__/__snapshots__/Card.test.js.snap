// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
.c0 {
  height: 100%;
  margin: 0 0 4%;
  border: 2px dashed;
  text-align: left;
  padding: 0 4%;
}

.c0 p {
  font-size: 1rem;
  line-height: 1.2;
}

.c0 .event-title {
  font-size: 1.2rem;
  text-transform: uppercase;
  font-family: sans-serif;
}

.c0 .event-title a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 .event-title span {
  font-weight: bold;
}

.c0 .event-cat {
  font-size: 0.9rem;
}

@media (max-width:480px) {
  .c0 span {
    display: none;
  }
}

<div
  class="c0"
/>
`;
